client/rest/src/plugins/rosetta/constructionRoutes.js

```diff
@@ -101,5 +123,231 @@ export default {
 			};
 			return response;
 		}));
+
+		const buildAddressToPublicKeyMap = rosettaPublicKeys => {
+			const addressToPublicKeyMap = {};
+			for (let i = 0; i < rosettaPublicKeys.length; ++i) {
+				const publicKey = parsePublicKey(rosettaPublicKeys[i]);
+				const address = facade.network.publicKeyToAddress(publicKey);
+				addressToPublicKeyMap[address.toString()] = publicKey;
+			}
+
+			return addressToPublicKeyMap;
+		};
+
+		const processTransfer = (addressToPublicKeyMap, subOperation1, subOperation2) => {
+			const makeDescriptor = (senderAccount, recipientAccount, amount) => ({
+				type: 'transfer_transaction_v1',
+				signerPublicKey: addressToPublicKeyMap[senderAccount.address],
+				recipientAddress: recipientAccount.address,
+				mosaics: [
+					{ mosaicId: generateMosaicAliasId('symbol.xym'), amount }
+				]
+			});
+
+			const amount = BigInt(subOperation1.amount.value);
+			if (0n !== amount + BigInt(subOperation2.amount.value))
+				throw RosettaErrorFactory.INVALID_REQUEST_DATA;
+
+			if (0n < amount)
+				return makeDescriptor(subOperation2.account, subOperation1.account, amount);
+
+			return makeDescriptor(subOperation1.account, subOperation2.account, -amount);
+		};
+
+		const createSigningPayload = (signingPayload, signerAddress) => {
+			const result = new SigningPayload(utils.uint8ToHex(signingPayload));
+			result.account_identifier = new AccountIdentifier(signerAddress.toString());
+			result.signature_type = 'ed25519';
+			return result;
+		};
+
+		const extractCosignerPublicKeys = (allSignerPublicKeys, aggregateSignerPublicKey) => {
+			const cosignerPublicKeyStringSet = new Set(allSignerPublicKeys.map(publicKey => publicKey.toString()));
+			cosignerPublicKeyStringSet.delete(aggregateSignerPublicKey.toString());
+			return [...cosignerPublicKeyStringSet].sort().map(publicKeyString => new models.PublicKey(publicKeyString));
+		};
+
+		server.post('/construction/payloads', rosettaPostRouteWithNetwork(networkName, ConstructionPayloadsRequest, async typedRequest => {
+			const addressToPublicKeyMap = buildAddressToPublicKeyMap(typedRequest.public_keys);
+
+			const embeddedTransactions = [];
+			const allSignerPublicKeys = [];
+			for (let i = 0; i < typedRequest.operations.length; ++i) {
+				const operation = typedRequest.operations[i];
+
+				if ('transfer' === operation.type) {
+					if (i + 1 === typedRequest.operations.length)
+						throw RosettaErrorFactory.INVALID_REQUEST_DATA;
+
+					const nextOperation = typedRequest.operations[i + 1];
+					if ('transfer' !== nextOperation.type)
+						throw RosettaErrorFactory.INVALID_REQUEST_DATA;
+
+					const embeddedTransaction = facade.transactionFactory.createEmbedded(processTransfer(
+						addressToPublicKeyMap,
+						operation,
+						nextOperation
+					));
+
+					embeddedTransactions.push(embeddedTransaction);
+					allSignerPublicKeys.push(embeddedTransaction.signerPublicKey);
+					++i;
+				} else if ('multisig' === operation.type) {
```

Comment:
Operation types must match what is returned by ``/network/options``, so we will need to add multisig and cosign.